### PR TITLE
Add Basic Elevated Message support

### PR DESCRIPTION
**Pull request by Felanbird**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Added Basic Elevated Message Support via the existing standard of First Messages.

While this approach obviously is missing the main concept of this feature which is the "pinned message", it allows for users to not be left out of the loop, and allows moderators to properly moderate said messages. 
`(e.g. A message that might have gotten past if it was a normal message may gain additional scrutiny if it is an Elevated Message)`

TODO: 
- [x] Figure out what I missed which is preventing `enableElevatedMessageHighlight` from disabling the scrollbar highlights. (currently `enableFirstMessageHighlight` is doing both)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 - Major: Added support for Twitch's Chat Replies. [Wiki Page](https://wiki.chatterino.com/Features/#message-replies) (#3722, #3989)
 - Major: Added multi-channel searching to search dialog via keyboard shortcut. (Ctrl+Shift+F by default) (#3694, #3875)
+- Minor: Added highlights for `Elevated Messages`. (#4016)
```
Comment (Felanbird) [inline CHANGELOG.md:7]: Changelog entry at the top of Minor due to recency bias.

File: src/controllers/highlights/HighlightPhrase.cpp [hunk 2]
```diff
@@ -14,6 +14,8 @@ QColor HighlightPhrase::FALLBACK_REDEEMED_HIGHLIGHT_COLOR =
     QColor(28, 126, 141, 60);
 QColor HighlightPhrase::FALLBACK_FIRST_MESSAGE_HIGHLIGHT_COLOR =
     QColor(72, 127, 63, 60);
+QColor HighlightPhrase::FALLBACK_ELEVATED_MESSAGE_HIGHLIGHT_COLOR =
+    QColor(255, 174, 66, 60);
```
Comment (Felanbird) [inline src/controllers/highlights/HighlightPhrase.cpp:18]: image
Open to suggestions, just picked the color from the `Priority:Medium` label and lowered the Alpha

File: src/messages/layouts/MessageLayout.cpp [hunk 3]
```diff
@@ -310,8 +310,16 @@ void MessageLayout::updateBuffer(QPixmap *buffer, int /*messageIndex*/,
...
+    if (this->message_->flags.has(MessageFlag::ElevatedMessage) &&
+        getSettings()->enableElevatedMessageHighlight.getValue())
+    {
+        backgroundColor = blendColors(backgroundColor,
+                                      *ColorProvider::instance().color(
+                                          ColorType::ElevatedMessageHighlight));
+    }
+
+    else if (this->message_->flags.has(MessageFlag::FirstMessage) &&
+             getSettings()->enableFirstMessageHighlight.getValue())
```
Comment (Felanbird) [inline src/messages/layouts/MessageLayout.cpp:322]: I went ahead and re-ordered priority so Elevated Messages are first. 
While knowing if a user's message is first is important, knowing if they paid money to elevate a stupid message is more important from a mod PoV

File: src/messages/Message.cpp [hunk 4]
```diff
@@ -48,6 +48,13 @@ SBHighlight Message::getScrollBarHighlight() const
             ColorProvider::instance().color(ColorType::FirstMessageHighlight),
             SBHighlight::Default, false, true);
     }
+    else if (this->flags.has(MessageFlag::ElevatedMessage))
+    {
+        return SBHighlight(ColorProvider::instance().color(
+                               ColorType::ElevatedMessageHighlight),
+                           SBHighlight::Default, false, true);
+    }
+
```
Comment (Felanbird) [inline src/messages/Message.cpp:57]: This area is the problematic code for the last issue
Currently the `enableFirstMessageHighlight` setting is disabling both First Messages and Elevated Messages.

Setting line 49 from `SBHighlight::Default, false, true);` to `SBHighlight::Default, true);` changed the issue to `enableFirstMessageHiglight` enable/disabling Elevated Messages scrollbar and `enableRedeemMessage` enable/disabling First Message scrollbar. `

File: src/singletons/Settings.hpp [hunk 5]
```diff
@@ -292,6 +292,17 @@ class Settings : public ABSettings, public ConcurrentSettings
...
+    BoolSetting enableElevatedMessageHighlight = {
+        "/highlighting/elevatedMessageHighlight/highlighted", true};
+    //    BoolSetting enableElevatedMessageHighlightSound = {
+    //        "/highlighting/elevatedMessageHighlight/enableSound", false};
+    //    BoolSetting enableElevatedMessageHighlightTaskbar = {
+    //        "/highlighting/elevatedMessageHighlight/enableTaskbarFlashing", false};
+    QStringSetting elevatedMessageHighlightSoundUrl = {
+        "/highlighting/elevatedMessageHighlightSoundUrl", ""};
+    QStringSetting elevatedMessageHighlightColor = {
+        "/highlighting/elevatedMessageHighlightColor", ""};
```
Comment (pajlada) [inline src/singletons/Settings.hpp:304]: Could you change the paths of these two to be under the `/highlighting/elevatedMessageHighlight/` "scope"? similar to the other 3 settings above (I know this is inconsistent with firstMessageHighlight)

**Review by pajlada — CHANGES_REQUESTED** [review]

File: src/singletons/Settings.hpp [hunk 6]
```diff
@@ -299,9 +299,9 @@ class Settings : public ABSettings, public ConcurrentSettings
     //    BoolSetting enableElevatedMessageHighlightTaskbar = {
     //        "/highlighting/elevatedMessageHighlight/enableTaskbarFlashing", false};
     QStringSetting elevatedMessageHighlightSoundUrl = {
-        "/highlighting/elevatedMessageHighlightSoundUrl", ""};
+        "/highlighting/elevatedMessageHighlight/SoundUrl", ""};
     QStringSetting elevatedMessageHighlightColor = {
-        "/highlighting/elevatedMessageHighlightColor", ""};
+        "/highlighting/elevatedMessageHighlight/Color", ""};
```
Comment (pajlada) [inline src/singletons/Settings.hpp:304]: camelCase `SoundUrl` to `soundUrl` and `Color` to `color`

**Review by pajlada — CHANGES_REQUESTED** [review]

**Review by pajlada — APPROVED** [review]
Looks good to me, I'm confident enough to have this merged in and then I'll test it more fully tomorrow with #4017 and #4018

**Pull request merged**